runners/js.py

```diff
@@ -0,0 +1,33 @@
+import subprocess, os
+from runners.python import Submission
+
+class SubmissionJs(Submission):
+
+	def __init__(self, file):
+		Submission.__init__(self)
+		self.file = file
+		self.script = """{script}
+result = run(process.argv[1]);
+console.log(result);
+""".format(script=open(file).read())
+
+	def language(self):
+		return 'js'
+
+	def run(self, s):
+		try:
+			output_raw = subprocess.check_output(["node", "-e", self.script, s]).decode()
+			output_rows = output_raw.split('\n')[:-1]
+			if len(output_rows) > 1:
+				print('\n'.join(output_rows[:-1]))
+			return output_rows[-1]
```

Comment:
About the protocol to transmit data from the subprocess to the main process: assuming there is only one output written on one single line, `Submission.run` returns the very last line in stdout of the subprocess (any other line is printed out as debug printing.)

Reply:
Nice approach, you can also add a `rstrip()` to eliminate any trailing spaces or line breaks.

Reply:
I'm not sure it's our role to deal with trailing characters. The dev should be allowed to output whatever she wants, formatting it how she intends, without having us interfering. We should only print what we are given, the way it is given.